### PR TITLE
slight refactorings, docs, typos

### PR DESCRIPTION
**Pull request by mikapfl**
@JGuetschow it was easier for me to add my refactorings as patches instead of doing them inline in github, so I created a PR into your branch with what I would change in your branch before merging. Changes:
* regenerated api docs (happens if you run "make docs" locally")
* added LocalTrendsStrategy to CSG usage docs
* added some docstrings, moved method docstrings from the class's docstring to the method's docstring
* unified calculate_right_boundary_trend and calculate_left_boundary_trend into one function with case distinctions
* refactored get_shifted_time_value. I first did it with python lists, which is a bit simpler, but using the underlying numpy arrays without converting is much faster, so I did that in the end.
* corrected typos wherever I found them.

### CODE REVIEW

**Comment by JGuetschow** [comment]
Thanks for adding docs. I should have said that I intentionally left them for after your review so I don't have to rewrite them in case of changes. Sorry, didn't intend to make you write the docs.

File: primap2/csg/_strategies/gaps.py [hunk 1]
```diff
@@ -12,23 +14,16 @@ class Gap:
 
     Attributes
     ----------
-    type :
+    type
```
Comment (JGuetschow) [inline primap2/csg/_strategies/gaps.py:17]: Seems that my pycharm setting for docstrings is not correct, that added the colons, I think

File: primap2/csg/_strategies/gaps.py [hunk 2]
```diff
@@ -71,15 +68,6 @@ class FitParameters:
         minimal number of points to calculate the trend. Default is 1, but if the degree
         of the fit polynomial is higher than 1, the minimal number of data points
         the degree of the fit polynomial
-
```
Comment (JGuetschow) [inline primap2/csg/_strategies/gaps.py:74]: So we use docstrings in the methods and not a methods section. I'll try to remember in the future.

**Review by JGuetschow — APPROVED** [review]

File: primap2/csg/_strategies/gaps.py (hunk 2 above)
Comment (mikapfl) [reply primap2/csg/_strategies/gaps.py:74]: Yeah, both works, but generally I like the docs to be as close to the code as possible.

**Comment by mikapfl** [comment]
@JGuetschow merge yourself if you are happy (will merge into your branch, not into main directly)

**Pull request merged**